rust/main/chains/hyperlane-sealevel/src/provider.rs

```diff
@@ -80,6 +91,39 @@ impl SealevelProvider {
         Ok(sender)
     }
 
+    fn recipient(hash: &H512, txn: &UiTransaction) -> ChainResult<H256> {
+        let message = Self::parsed_message(txn)?;
+
+        let programs = message
+            .instructions
+            .iter()
+            .filter_map(|ii| {
+                if let UiInstruction::Parsed(iii) = ii {
+                    Some(iii)
+                } else {
+                    None
+                }
+            })
+            .map(|ii| match ii {
+                UiParsedInstruction::Parsed(iii) => &iii.program_id,
+                UiParsedInstruction::PartiallyDecoded(iii) => &iii.program_id,
+            })
+            .filter(|program_id| !NATIVE_PROGRAMS.contains(*program_id))
+            .collect::<Vec<&String>>();
+
+        if programs.len() > 1 {
+            Err(HyperlaneSealevelError::TooManyNonNativePrograms(*hash))?;
+        }
+
+        let program_id = programs
+            .first()
+            .ok_or(HyperlaneSealevelError::NoNativePrograms(*hash))?;
```

Comment:
any thoughts on how likely this is to occur? if you think there's a possibility, wondering if we should just put a dummy value in this case or something